### PR TITLE
refactor: float conversion methods

### PR DESCRIPTION
**Pull request by aljo242**
Closes BLO-860

- introduce `BigFloatToBigInt` which is used by `Float64StringToBigInt`
- this means that float64 strings are not converted in a way where any precision is lost
- add more tests for the package 
- some formatting changes from running `make format`

### CODE REVIEW

**Review by davidterpay — APPROVED** [review]

File: pkg/math/math.go [hunk 1]
```diff
@@ -11,40 +10,44 @@ func Min[V int | int64 | uint64 | int32 | uint32](vals ...V) V {
...
 		return nil, err
 	}
 
-	return Float64ToBigInt(floatNum, decimals), nil
+	return BigFloatToBigInt(bigFloat, decimals), nil
 }
 
 // Float64ToBigInt converts a float64 to a big.Int.
-//
-// TODO: Is there a better approach to this?
```
Comment (g3ndo9kar9) [inline pkg/math/math.go:35]: Where is this being used?

File: pkg/math/math.go [hunk 2]
```diff
@@ -11,40 +10,44 @@ func Min[V int | int64 | uint64 | int32 | uint32](vals ...V) V {
...
+
+// BigFloatToBigInt converts a big.Float to a big.Int.
+func BigFloatToBigInt(f *big.Float, decimals int) *big.Int {
+	bigFloat := new(big.Float)
 	factor := big.NewInt(1).Exp(big.NewInt(10), big.NewInt(int64(decimals)), nil)
-	coin.SetInt(factor)
+	bigFloat.SetInt(factor)
 
-	bigval.Mul(bigval, coin)
+	f.Mul(f, bigFloat)
```
Comment (g3ndo9kar9) [inline pkg/math/math.go:47]: Maybe add a benchmark or smth? Given how frequently this is used, we shld at least have some introspection into perf

**Review by g3ndo9kar9 — APPROVED** [review]
👑

File: pkg/math/math.go (hunk 2 above)
Comment (aljo242) [reply pkg/math/math.go:47]: added benchmarks and new `make test-bench` target

**Pull request merged**